facilities/models.py

```diff
@@ -0,0 +1,83 @@
+import random
+from django.db import models
+from common.models import AbstractBase, SubCounty
+
+
+class Owner(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def generate_code(self):
+        return "{}{}".format("OWNER", str(self.id))
+
+    def save(self, *args, **kwargs):
+        if not self.code:
+            self.code = self.generate_code()
+        super(Owner, self).save(*args, **kwargs)
+
+
+class Service(AbstractBase):
+    name = models.CharField(max_length=255)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, unique=True)
+
+    def __unicode__(self):
+        return self.name
+
+    def generate_code(self):
+        return "{}{}".format(self.name, str(self.id))
+
+    def save(self, *args, **kwargs):
```

Comment:
You will not require a custom `save` method when you move to a `default=` callable.